crates/language/src/language_settings.rs

```diff
@@ -563,7 +563,7 @@ impl settings::Settings for AllLanguageSettings {
             .and_then(|c| c.disabled_globs.as_ref())
             .ok_or_else(Self::missing_default)?;
 
-        let mut file_types: HashMap<Arc<str>, Vec<String>> = HashMap::default();
+        let mut file_types: Option<HashMap<Arc<str>, GlobSet>> = Some(HashMap::default());
```

Comment:
```suggestion
        let mut file_types: HashMap<Arc<str>, GlobSet> = HashMap::default();
```
I'm pretty sure that we may be able to omit the type altogether, though I can't check that rn:
```suggestion
        let mut file_types = HashMap::default();
```